### DISCUSSION TITLE
Android Platform 10.1.2 - Build failed for Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0

### DISCUSSION THREAD

**Discussion by FranGhe** [HaylLtd/cordova-plugin-background-geolocation#114] [Q&A]
Just 1 month ago I have published my first App on PlayStore using cordova-plugin-mauron85-background-geolocation

Now I have to fix small bugs, but when I start the build it fails... searching the solution on the web I found this topic that match with mine:
https://github.com/mauron85/cordova-plugin-background-geolocation/issues/766

Following the suggestion I have decided to remove that plugin and add your... but now I have now another issue... When I make the build it fail with this message:
```
Execution failed for task ':app:compileDebugJavaWithJavac'
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0
```
 if I remove your plugin the build is successful... I am absolutely sure that Gradle is installed when I install the Android platform... and it is out of my control.

These are my versions:
```
Cordova: 10.0.0
Android Platform: 10.1.2 
Gradle: 7.1.1
```

And these are my plugins:
```
cordova-background-geolocation-plugin 2.0.7
cordova-plugin-androidx-adapter 1.1.3
cordova-plugin-app-version 0.1.14
cordova-plugin-device 2.1.0
cordova-plugin-dialogs 2.0.2
cordova-plugin-file 7.0.0
cordova-plugin-geolocation 4.1.0
cordova-plugin-globalization 1.11.0
cordova-plugin-inappbrowser 5.0.0
cordova-plugin-insomnia 4.3.0
cordova-plugin-native-app-update 2.0.0
cordova-plugin-network-information 3.0.0
cordova-plugin-splashscreen 6.0.2
cordova-plugin-statusbar 3.0.0
```

How I can fix the issue?

**Comment by HarelM** [comment]
The following is not the issue as it is only a warning.
`Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0`

Also note that you have both `cordova-plugin-geolocation` and `cordova-background-geolocation-plugin` which I think can cause issues.

I have this plugin in my app so I know it complies in Android and works well.
Please share more info.

Converting this to a discussion.

**Comment by FranGhe** [comment]
@HarelM 

I don't know what to say... for me it's a issue because cordova can't build the project and install it on my smartphone.

What version of of Cordova, Android and Gradle do you have?

If I unistall cordova-plugin-geolocation and try again the build ... it doesn't works again.

When Cordova build the project, before to write the message `Build failed for Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0` write this:

```
> Task :app:compileDebugJavaWithJavac FAILED
C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\java\com\marianhello\bgloc\provider\DistanceFilterLocationProvider.java:88: error: cannot find symbol
        int zeroFlag = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
                                                                   ^
  symbol:   variable S
  location: class android.os.Build.VERSION_CODES
C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\java\com\marianhello\bgloc\provider\DistanceFilterLocationProvider.java:89: error: cannot find symbol
            ? 0 | PendingIntent.FLAG_MUTABLE
                               ^
  symbol:   variable FLAG_MUTABLE
  location: class android.app.PendingIntent
C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\java\com\marianhello\bgloc\provider\DistanceFilterLocationProvider.java:92: error: cannot find symbol
        int cancelCurrentFlag = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
                                                                            ^
  symbol:   variable S
  location: class android.os.Build.VERSION_CODES
C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\java\com\marianhello\bgloc\provider\DistanceFilterLocationProvider.java:93: error: cannot find symbol
            ? PendingIntent.FLAG_CANCEL_CURRENT | PendingIntent.FLAG_MUTABLE
                                                               ^
  symbol:   variable FLAG_MUTABLE
  location: class android.app.PendingIntent
C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\java\com\marianhello\bgloc\sync\NotificationHelper.java:81: error: cannot find symbol
                int flags = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
                                                                        ^
  symbol:   variable S
  location: class android.os.Build.VERSION_CODES
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
5 errors
```

If I uninstall `cordova-background-geolocation-plugin` I have this messages:
```
Uninstalling cordova-background-geolocation-plugin from android
config file res/xml/authenticator.xml requested for changes not found at C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\res\xml\authenticator.xml, ignoring
config file res/xml/syncadapter.xml requested for changes not found at C:\NetBeansProjects\App-PaeseMio\platforms\android\app\src\main\res\xml\syncadapter.xml, ignoring
```

I hope that all of this can explain my problem.

Thanks for your support

**Reply by ivinantony** [reply]
this is the same error i am facing too

**Comment by HarelM** [comment]
Use higher android target SDK or previous version of the plugin.

**Reply by FranGhe** [reply]
About android target SDK I had this on my config.xml
`<preference name="android-targetSdkVersion" value="30" />`

I have to set it more or previous plugin? ... in last case ... which version?

**Reply by ivinantony** [reply]
but if we use higher 31,32 it gives an error of android:exported="true". i am still stuck on this

**Comment by ivinantony** [comment]
<img width="1415" alt="Screenshot 2022-09-29 at 9 47 01 PM" src="https://user-images.githubusercontent.com/66516836/193084656-7fc9db1c-3f83-41f0-91c8-3d0f02431231.png">
